core-bundle/src/Doctrine/Schema/DcaSchemaProvider.php

```diff
@@ -316,7 +313,20 @@ private function parseIndexSql(Table $table, $keyName, $sql)
                 $flags[] = 'fulltext';
             }
 
-            $table->addIndex($columns, $matches[2], $flags);
+            $options = [];
+
+            if (array_sum($lengths)) {
+                $options['lengths'] = $lengths;
+
+                // Backwards compatibility for doctrine/dbal < 2.9
+                if (!method_exists(AbstractPlatform::class, 'supportsColumnLengthIndexes')) {
+                    $columns = array_combine($columns, array_map(function($column, $length) {
+                        return $column . '(' . $length . ')';
```

Comment:
Shouldn't we check `null !== $length` here?

Reply:
Done in c714532aa25394b60cd75790bffc6e559be56d7b